src/providers/https.ts

```diff
@@ -163,88 +195,44 @@ export class HttpsError extends Error {
    * A standard error code that will be returned to the client. This also
    * determines the HTTP status code of the response, as defined in code.proto.
    */
-  readonly code: FunctionsErrorCode;
+  public readonly code: FunctionsErrorCode;
 
   /**
    * Extra data to be converted to JSON and included in the error response.
    */
-  readonly details?: unknown;
+  public readonly details: unknown;
+
+  /**
+   * A wire format representation of a provided error code.
+   */
+  public readonly httpErrorCode: HttpErrorCode;
 
   constructor(code: FunctionsErrorCode, message: string, details?: unknown) {
     super(message);
 
-    if (!errorCodeMap[code]) {
-      throw new Error('Unknown error status: ' + code);
+    // A sanity check for non-TypeScript consumers.
+    if (code in errorCodeMap === false) {
+      throw new Error(`Unknown error code: ${code}.`);
     }
 
     this.code = code;
     this.details = details;
-  }
-
-  /**
-   * @hidden
-   * A string representation of the Google error code for this error for HTTP.
-   */
-  get status() {
-    return errorCodeMap[this.code];
-  }
-
-  /**
-   * @hidden
-   * Returns the canonical http status code for the given error.
-   */
-  get httpStatus(): number {
-    switch (this.code) {
-      case 'ok':
-        return 200;
-      case 'cancelled':
-        return 499;
-      case 'unknown':
-        return 500;
-      case 'invalid-argument':
-        return 400;
-      case 'deadline-exceeded':
-        return 504;
-      case 'not-found':
-        return 404;
-      case 'already-exists':
-        return 409;
-      case 'permission-denied':
-        return 403;
-      case 'unauthenticated':
-        return 401;
-      case 'resource-exhausted':
-        return 429;
-      case 'failed-precondition':
-        return 400;
-      case 'aborted':
-        return 409;
-      case 'out-of-range':
-        return 400;
-      case 'unimplemented':
-        return 501;
-      case 'internal':
-        return 500;
-      case 'unavailable':
-        return 503;
-      case 'data-loss':
-        return 500;
-      // This should never happen as long as the type system is doing its job.
-      default:
-        throw new Error('Invalid error code: ' + this.code);
-    }
+    this.httpErrorCode = errorCodeMap[code];
   }
 
   /** @hidden */
-  public toJSON() {
-    const json: any = {
-      status: this.status,
-      message: this.message,
+  public toJSON(): HttpErrorWireFormat {
```

Comment:
Hm, just noticing that having public and @hidden together seems a little odd. @egilmorez do we expose toJSON() method on https provider? I see we do in other providers but I didn't find it for https.

Reply:
I applied this change in fe67bd7ab994ceb4dfd43416ccd8e35fb26be2b9 so we can merge if @egilmorez agrees. I'll revert otherwise.

Reply:
Let's hide this for now.